BeeSwift/CurrentUserManager.swift

```diff
@@ -21,79 +21,119 @@ class CurrentUserManager : NSObject {
     static let goalsFetchedNotificationName = "com.beeminder.goalsFetchedNotification"
     static let healthKitMetricRemovedNotificationName = "com.beeminder.healthKitMetricRemovedNotification"
     
+    fileprivate let beemiosSecret = "C0QBFPWqDykIgE6RyQ2OJJDxGxGXuVA2CNqcJM185oOOl4EQTjmpiKgcwjki"
+    
     fileprivate let accessTokenKey = "access_token"
     fileprivate let usernameKey = "username"
     fileprivate let deadbeatKey = "deadbeat"
     fileprivate let defaultLeadtimeKey = "default_leadtime"
     fileprivate let defaultAlertstartKey = "default_alertstart"
     fileprivate let defaultDeadlineKey = "default_deadline"
     fileprivate let beemTZKey = "timezone"
-    fileprivate let beemiosSecret = "C0QBFPWqDykIgE6RyQ2OJJDxGxGXuVA2CNqcJM185oOOl4EQTjmpiKgcwjki"
+    
+    var allKeys: [String] {
+        return [accessTokenKey, usernameKey, deadbeatKey, defaultLeadtimeKey, defaultAlertstartKey, defaultDeadlineKey, beemTZKey]
+    }
+    
+    let userDefaults = UserDefaults(suiteName: Constants.appGroupIdentifier)!
+    
+    override init() {
+        super.init()
+        migrateValues()
+    }
+    
+    /// Migrate settings values from the standard store to a group store
+    ///
+    /// Originally BeeSwift stored all configuration values in the standard UserDefaults store. However
+    /// these values are not available within extensions. To address this now values are stored in a
+    /// group-scoped settings object. Values written by old versions of the app may be in the previous store
+    /// so we migrate any such values on initialization.
+    private func migrateValues() {
+        for key in allKeys {
```

Comment:
How about

```suggestion
   allKeys.forEach ...
```

Reply:
I agree this is also correct. Are there reasons to choose one form over the other? (Googling didn't turn up anything compelling.) The rest of the codebase seems to mainly, but not entirely, use the `.forEach` form.

Reply:
In this case I prefer the latter since appears to match more closely the functional programming style.